### PR TITLE
refactor: Node and FormData

### PR DESCRIPTION
**Pull request by cjmarkham**
## Type

* ### Fix
  Fixes a bug

## Description

Node users using an implementation of FormData are hampered by the fact that the SDK doesn't set the Content-Type header. For the browser this isn't an issue. 
We can't rely on checking for an instance of FormData because there's many implementations.
The solution is to allow the user to specify an additional parameter on `Files.Create` to specify the content type

A `form-data` implementation:
```
Moltin.Files.Create(form, form.getHeaders())
```

Additionally, the `Create` method has been split up. In order to create a file with only a file_location, there's a new `Link` method:

```
Moltin.Files.Link(href: string)
```

## Dependencies

None

## Issues

None

### CODE REVIEW

File: src/endpoints/files.js [hunk 1]
```diff
@@ -8,26 +9,38 @@ class Files extends BaseExtend {
...
+      undefined,
       undefined,
+      false,
       undefined,
-      false
+      additionalHeaders
     )
   }
 
+  Link(href) {
```
Comment (cjmarkham) [inline src/endpoints/files.js:33]: `href` is Hypertext REFerence, so I would probably rename this to url or something instead.

File: src/endpoints/files.js [hunk 2]
```diff
@@ -8,26 +9,38 @@ class Files extends BaseExtend {
     this.endpoint = 'files'
   }
 
-  Create(body) {
-    let requestBody = body
+  Create(body, contentType = null) {
+    const additionalHeaders = {}
```
Comment (cjmarkham) [inline src/endpoints/files.js:13]: We could do with a check here in the event that a node env doesn't specify the header
```
if (isNode() && contentType === null) throw error
```

File: src/endpoints/files.js (hunk 1 above)
Comment (samblacklock) [reply src/endpoints/files.js:33]: agree

File: src/endpoints/files.js (hunk 2 above)
Comment (samblacklock) [reply src/endpoints/files.js:13]: yeh thats a good idea 👍

**Review by samblacklock — APPROVED** [review]

**Pull request merged**